### PR TITLE
Added Error Bar Composite Mark

### PR DESCRIPTION
**Pull request by ayush29feb**
TODO
- [ ] normalize `composite unitSpec` in `LayerSpec`.`layers`
- [x] copy `size` to ticks

### CODE REVIEW

File: src/spec.ts [hunk 1]
```diff
@@ -161,6 +168,42 @@ export function normalizeUnitSpec(spec: UnitSpec): Spec {
   return spec;
 }
 
+export function normalizeCompositeUnitSpec(spec: UnitSpec): Spec {
+  let layerSpec = extend(spec.name ? {name: spec.name} : {},
+    spec.description ? {description: spec.description} : {},
+    spec.data ? {data: spec.data} : {},
+    spec.transform ? {transform: spec.transform} : {},
+    spec.config ? {config: spec.config} : {}, {layers: []});
```
Comment (kanitw) [inline src/spec.ts]: break line before the last `);`
and break line before starting the if clause.

File: src/spec.ts [hunk 2]
```diff
@@ -161,6 +168,42 @@ export function normalizeUnitSpec(spec: UnitSpec): Spec {
...
+  if (!spec.encoding) {
+    return normalize(layerSpec);
```
Comment (kanitw) [inline src/spec.ts]: I don't really know why you need this clause.  Is it necessary?

File: src/spec.ts [hunk 3]
```diff
@@ -139,6 +139,13 @@ export function normalizeExtendedUnitSpec(spec: ExtendedUnitSpec): Spec {
 }
 
 export function normalizeUnitSpec(spec: UnitSpec): Spec {
+  if (contains([ERRORBAR], spec.mark)) {
+    return normalizeCompositeUnitSpec(spec);
+  }
+  return normalizeRangedUnitSpec(spec);
```
Comment (kanitw) [inline src/spec.ts]: If would put the condition to check if this spec is range here -- for consistency.  (so `normalizeRangedUnitSpec` only transform the spec)

**Comment by kanitw** [comment]
- [ ] (later) unit test

**Comment by kanitw** [comment]
- [ ] unit test (later)

File: src/spec.ts (hunk 2 above)
Comment (ayush29feb) [inline src/spec.ts]: Wasn't sure if you can `duplicate` `undefined`

Comment (kanitw) [inline src/spec.ts]: Good point.  But why do we need to call `normalize` another time?

Comment (kanitw) [inline src/spec.ts]: I think if there is no `encoding`, it shouldn't even enter this method from the beginning.

Comment (kanitw) [inline src/spec.ts]: We can make sure `isUnitSpec` check for `spec.encoding`.

Comment (ayush29feb) [inline src/spec.ts]: Okay.. that works better.

**Comment by ayush29feb** [comment]
@kanitw @domoritz should be allow composite mark types to be used within Layers? I am guessing yes. Can either of you point me where does each `unitSpec` is being `normalize`'d?

**Pull request merged**